### ISSUE TITLE
Intel tbb win support

### ISSUE THREAD

**Issue by mintyc** [preshing/turf#2]
Recently found that visual studio 2015 benefitted significantly from using the tbb malloc alternative.

Any support likely in turf?

The TL:DR bit

Other allocators claiming good multithread performance did not scale well with modest numbers of threads.
E.g 4 cores each with 10m individually heap allocated 450 byte objects.
Simultaneous map resets took 8 sec per core and 9 secs wall clock time.

Alternative allocators below had some variation but typically became so bad the best of the others was 11sec one core, 50 sec to complete four cores. Serialised and extra lock contention.

Note the test is not a great design and should be using pool allocation but it illustrates the huge differences in windows multithreaded allocator support.

I cant vouch for the rest of tbb but the allocator scaled well when tcmalloc jemalloc and hoard all failed to scale in vs2015.
I suspect the others lag behind because they may not yet implement the latest hooks needed in the new crt.

**Comment by mintyc** [comment]
I mean in addition to me doing a regular replacement of the allocators as per tbb recommendations.
Explicit support may be able to optimise some of the low level calls?

**Comment by preshing** [comment]
If I understand correctly, you're suggesting a configuration where all calls to TURF_HEAP.alloc/free get mapped to an allocator in one of the TBB DLLs, right?

What TBB functions should they get mapped to, exactly? I don't have the docs handy at the moment.

**Comment by preshing** [comment]
Let's say there are no plans at this time, but contributions welcome.
